src/lib/Scenes/Collection/Collection.tsx

```diff
@@ -60,19 +67,66 @@ export class Collection extends Component<CollectionProps, CollectionState> {
           </Box>
         )
       case "collectionArtworks":
-        return <CollectionArtworks collection={this.props.collection} />
+        return (
+          <>
+            <CollectionArtworks collection={this.props.collection} />
+            <FilterModal
+              visible={this.state.isFilterArtworksModalVisible}
+              closeModal={this.handleFilterArtworksModal.bind(this)}
+            />
+          </>
+        )
       default:
         return null
     }
   }
+  onViewableItemsChanged = ({ viewableItems }) => {
+    ;(viewableItems || []).map(viewableItem => {
+      const artworksRenderItem = viewableItem?.item?.type || ""
```

Comment:
A #minor comment (feel free to punt until a later PR!) but the variable naming here is a little bit confusing (`artworksRenderItem` feels a little too specific given we may or may not be handling the artwork filter).

Something like 
```ts
const itemType = viewableItem?.item?.type || ""
const itemIsVisible = viewableItem?.isViewable || false

if (itemType === "collectionsArtworks" && itemIsVisible)..
```

could be clearer!